<launch>
    <node name="turtle_yaw_pid" pkg="pid" type="controller" output="screen" >
      <param name="Kp" value="0.02" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.01" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="2.84" />
      <param name="cutoff_frequency" value="20" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="1.0" />
      <param name="setpoint_timeout" value="-1.0" />
      <!-- custom  -->
      <param name="topic_from_controller" value="/pid/out_angvel" />
      <param name="topic_from_plant" value="/razor/yaw" />
      <param name="setpoint_topic" value="/pid/target_yaw" />
      <param name="pid_enable_topic" value="/pid/enabled" />
      <param name="angle_error" value="true" />
      <param name="angle_wrap" value="360" />
    </node>

    <node name="turtle_yaw_node" pkg="pid" type="plant_sim" output="screen" >
      <param name="plant_order" value="2" />
    </node>

    <node name="setpoint_node" pkg="pid" type="setpoint_node" output="screen" />

    <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <!-- <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/pid/out_angvel/data /razor/yaw/data /pid/target_yaw/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" /> !-->
</launch>
